### ISSUE TITLE
nosleep.enable() not working on bookmark saved as an application in iphone

### ISSUE THREAD

**Issue by tazo-taz** [richtr/NoSleep.js#161]
* Device Name: iPhone
* Device Version: 11
* Browser Name: Safari
* Browser Version: latest
* NoSleep.js Version: ^0.12.0

### Expected Behavior
The noSleep.enable() function should prevent the screen from sleeping when the web application is launched from a bookmark saved as an application on an iPhone.

### Actual Behavior
The noSleep.enable() function does not prevent the screen from sleeping when the web application is launched from a bookmark.

### Code
```javascript
useEffect(() => {
    const noSleep = new NoSleep();
    const documentOnClick = async () => {
      try {
        const a = await noSleep.enable();
        console.debug("no sleep enabled", a);
      } catch (e) {
        console.debug("no sleep prevented");
      }
    };
    document.addEventListener("click", documentOnClick);
    return () => {
      document.removeEventListener("click", documentOnClick);
      noSleep.disable();
    };
  }, []);

**Comment by av01d** [comment]
This is a known bug on iOS: https://bugs.webkit.org/show_bug.cgi?id=254545

**Comment by Finesse** [comment]
[The video method](https://github.com/richtr/NoSleep.js/blob/1d15cc88dd8c1b62fe5b5f910c6ff7396b280d68/src/index.js#L37) works both in PWA and Safari on iOS 17. But the library doesn't use it because [navigator.wakeLock](https://developer.mozilla.org/en-US/docs/Web/API/WakeLock) is available. So the library maintainer should either use both `navigator.wakeLock` and video, or provide an option for us to choose.

As user, you can copy the library code to your website and modify it so both methods are used.

**Comment by tazo-taz** [comment]
@Finesse thank you for response, I tried that by break if else if else and it will enter into both condition but still has same problem on PWA any more ideas to fix or did you manage you to fix that

**Comment by Finesse** [comment]
@tazo-taz No, I failed to make it work reliably. I chose to use my app in Safari, because wake lock works more reliably there. Though it requires user action to activate, so I made an infinite loop waiting for any user action, then waiting for a lock release, then waiting for user action and so on.
